### ISSUE TITLE
wrong printing for single-element matrix

### ISSUE THREAD

**Issue by simeonschaub** [JuliaLang/julia#36732]
The following printing is wrong, as the output would produce a vector, not a matrix:
```julia
julia> show(IOContext(stdout, :compact=>true), fill(42, 1, 1))
[42]
```
Observed on master and 1.5.

**Comment by StefanKarpinski** [comment]
The trouble is we don't have an input syntax for single-column matrices.

**Comment by simeonschaub** [comment]
We could probably just print this as `fill(42, 1, 1)`. I don't think there is a more concise way to otherwise express this.

**Comment by JeffBezanson** [comment]
Related: #28905

**Comment by simeonschaub** [comment]
@thofma and @mbauman seemed against this in #36737, but I still think it would be good to be able to distinguish matrices from vectors when printed inside arrays. Perhaps a compromise would be printing matrices with a trailing semicolon, for example `[42;]`, although this would of course still produce a vector, so might be more confusing.

**Comment by simeonschaub** [comment]
This is now fixed by #33697
